### ISSUE TITLE
Bug editing header of collapsed method

### ISSUE THREAD

**Issue by andy-ms** [Microsoft/vscode#25350]
- VSCode Version: 1.12.0-insider
- OS Version: Linux Mint 18

Steps to Reproduce:

1. Paste this code:
```ts
class C {
    m() {
        if (true) {
            return;
        }
    }
}
```

2. `Ctrl-K`, `Ctrl-0` to collapse all.
3. Expand `class C`. (Do not expand the `m`.)
4. Click inside the parens in `m` and write `param: number`.
5. An extra `...` symbol appears in the middle of the inserted text.

![screenshot](https://cloud.githubusercontent.com/assets/19274678/25410459/c2909e1c-29cb-11e7-8af1-d80112772460.jpg)

**Comment by aeschli** [comment]
The bug is related to the view model mapping. Hidden markers are mapped to the end of the visible line, but not updated if that line is edited.
- create the folloing typescript file
- collapse m(..): Squiggly is at the end of the line
- type inside the comment: Squiggly is no longer at the end of the line

```typescript
class C {
    m(/**/){
        if (true1) {
            return;
        }
    }
}
```

**Comment by alexandrudima** [comment]
This has been fixed in the meantime via #36410
